script.js

```diff
@@ -1,4 +1,238 @@
+//There are 2 players and they take turns to play
+//When 1 player clicks submit, 2 dices roll at once.
+//The player can combine these dice numbers to the order they like.
+//I specify how they order
+//The higher combo wins
+
+//EG. GREYBOX OUTPUT1: Welcome Player 1 / You rolled 3 for Dice 1 and  6 for Dice 2. / Choose the order of your dice
+//EG. GREYBOX OUTPUT2:
+//player 1, you chose Dice 2 first. / Your combo number is 63. / It is now Player 2 's turn to roll the dice.
+
+//GAME MODIFICATION TO PERSONALISE AND MAKE GAME MORE FUN!!
+//ADDED USERNAMES INPUT FOR PLAYER1 AND PLAYER2 (Lines42  - 74)
+//HIDE THE DICE VALUES UNTIL ORDER OF DICE IS CHOSEN.
+//PLAYERS ARE GIVEN A RUNNING LIST OF THEIR RANDOM DICEROLLS
+///////////////////////////////////////////////////////////////////
+var currentGameMode = "player1 name input";
+var playerName1 = "";
+var playerName2 = "";
+var playerName1Dices = [];
+var playerName2Dices = [];
+var playerName1DicesLog = [];
+var playerName2DicesLog = [];
+var playerName1Order = "";
+var playerName2Order = "";
+
+//var playerName1Counter = 0;
+//var playerName2Counter = 0;
+
+///////////////////////////////////////////////////////////////////
+var rollDice = function () {
+  var randomDice1 = Math.floor(Math.random() * (6 - 1 + 1) + 1);
+  return randomDice1;
+};
+
+///////////////////////////////////////////////////////////////////
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+
+  //Player1 enters name
+  if (currentGameMode == "player1 name input") {
+    myOutputValue = "Hello Player1, please enter your name";
+    currentGameMode = "player2 name input";
+    return myOutputValue;
+  }
+  if (currentGameMode == "player2 name input") {
+    playerName1 = input;
+    myOutputValue =
+      "Hello " +
+      input +
+      "!" +
+      "<br>" +
+      " It's now Player2's turn to enter his/her name!";
+    currentGameMode = "The Game Starts";
+    return myOutputValue;
+  }
+
+  //Player2 enters name
+  if (currentGameMode == "The Game Starts") {
+    playerName2 = input;
+    myOutputValue =
+      "Hello " +
+      input +
+      "!" +
+      "<br>" +
+      " It's now " +
+      playerName1 +
+      "'s turn to start the dice rolling! 2 secret dices will roll with each play! " +
+      "<br>" +
+      playerName1 +
+      "  , Please enter 1,2 for original order or 2,1 for reverse order for the secret dice rolls and click `submit` once";
+    currentGameMode = "Dice Roll Player1";
+    return myOutputValue;
+  }
+
+  //The Game starts after both players identify themselves.
+
+  if (currentGameMode == "Dice Roll Player1") {
+    var playerName1RandomDice1 = rollDice();
+    var playerName1RandomDice2 = rollDice();
+    playerName1Dices = [playerName1RandomDice1, playerName1RandomDice2];
+    //.push to array for Player1 to contain all his dices' numbers.
+    playerName1DicesLog.push(playerName1Dices);
+
+    var playerName1DiceChoose = input;
```

Comment:
More comments would be good to explain the intention behind what you are doing here. Additionally, consider using variable names that better reflect your intention as well.  For example, in these lines,  what is playerName1DicesLog supposed to be? Why is it an array of arrays? <-- these are questions that you could try to answer/explain using comments